pkg/database/db_interface.go

```diff
@@ -1,16 +1,18 @@
 package database
 
 import (
+	"context"
+
 	"github.com/Ubivius/microservice-template/pkg/data"
 )
 
 // The interface that any kind of database must implement
 type ProductDB interface {
-	GetProducts() data.Products
-	GetProductByID(id string) (*data.Product, error)
-	UpdateProduct(product *data.Product) error
-	AddProduct(product *data.Product) error
-	DeleteProduct(id string) error
+	GetProducts(ctx context.Context) data.Products
+	GetProductByID(ctx context.Context, id string) (*data.Product, error)
+	UpdateProduct(ctx context.Context, product *data.Product) error
+	AddProduct(ctx context.Context, product *data.Product) error
+	DeleteProduct(ctx context.Context, id string) error
 	Connect() error
```

Comment:
Not sure if I should pass the context.Background() to connect, ping and close instead of just calling it directly in the functions @MaximeStOnge @JeremiSavard